### DISCUSSION TITLE
Adds extensios for ProcessBuilder for Null or Empty string checks.

### DISCUSSION THREAD

**Discussion by phillipsj** [cake-build/cake#3345] [Ideas]
I find myself writing this code a lot:

```
var builder = new ProcessArgumentBuilder();

if(!string.IsNullOrWhitespace(settings.MySetting){
    builder.Append(settings.MySetting);
}
```
I would like to add the following extension method or method to the ProcessArgumentBuilder to make this scenario easier.

```
public static ProcessArgumentBuilder AppendIfNotNullOrWhitespace(this ProcessArgumentBuilder builder, string setting){
    return !string.IsNullOrWhitespace(setting)? builder.Append(setting) : builder;
}
```

Then the code would look like this:

```
var builder = new ProcessArgumentBuilder();
builder.AppendIfNotNullOrWhitespace(settings.MySetting);
```

I think the name could be made shorter.

**Comment by mvput** [comment]
Is this up for grabs?

**Comment by devlead** [comment]
Don't think @cake-build/cake-team has looked at this one yet.

This possibly misses the scenario where it often is a key value pair and you in many cases want to omit the key too.
Perhaps something `builder.IfNotNullOrWhitespace(settings?.MySetting?.ToString(), (pb, text)=>pb.AppendSwitchSecret("MySetting", text)`
via something like `public static ProcessArgumentBuilder IfNotNullOrWhitespace(this ProcessArgumentBuilder builder, string value,  Func<ProcessArgumentBuilder, T, ProcessArgumentBuilder> func)`
could handle that scenario.

**Comment by patriksvensson** [comment]
I agree with @devlead here. It will be too many variations of this otherwise.
